File: 05- May/02- Sign of the Product of an Array/02- Sign of the Product of an Array (Ibrahim Khalid).cpp
// Author : Ibrahim Khalid 
class Solution {
public:
    int arraySign(vector<int>& nums) {
        // to count negative numbers
         int neg=0;
        for(int i=0;i<nums.size();i++){
            // if equal 0 return 0;
          if(nums[i]==0){
              return 0;
          }
          else if(nums[i]<0){
              neg++;
          }
        }
        // if numbers of negative numbers is even then x is positive 
        if(neg%2==0){
            return 1;
        }
    // else 
        return -1;
    }
};

File: 05- May/02- Sign of the Product of an Array/02- Sign of the Product of an Array (Ibrahim Khalid).py
# Author : Ibrahim Khalid 
class Solution(object):
    def arraySign(self, nums):
        # to count negative numbers
        cnt_neg=0
        for x in nums:
            #  if equal 0 return 0;
            if(x==0):
                return 0
            if(x<0):
                cnt_neg+=1
        #if numbers of negative numbers is even then x is positive
        if( cnt_neg%2==0):
            return 1
        return -1
       
